plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -681,6 +681,30 @@ public static function current_screen( $screen ) {
 		}
 	}
 
+	/**
+	 * Get helper redirect URL.
+	 * @param array $args Query args.
+	 * @return string
+	 */
+	private static function _get_helper_redirect_url( $args = array() ) {
+		global $current_screen;
+		if ( isset( $_GET['redirect-to-wc-admin'] ) && $current_screen->id === 'woocommerce_page_wc-addons' ) {		
+			return add_query_arg(
+				array(
+					'page' => 'wc-admin',
+					'tab'  => 'my-subscriptions',
+					'path' => urlencode( '/extensions' ),
+				),
+				admin_url( 'admin.php' )
+			);
+		}
+		
+		return add_query_arg(
+			$args,
+			admin_url( 'admin.php' )
+		);
```

Comment:
I find it a little confusing that you instruct this function to redirect you to the `wc-addons` page when calling it:

```php
$redirect_uri = self::_get_helper_redirect_url(
			array(
				'page'             => 'wc-addons',
				'section'          => 'helper',
				'wc-helper-status' => 'helper-disconnected',
			)
		);
```

But the function may ignore your instruction by looking up a global variable (`$_GET['redirect-to-wc-admin']`) and send you somewhere else.

There is nothing wrong with looking at the `$_GET` variable, however, reading the code may confuse us later. 

I recommend renaming the function to `get_subscriptions_page_url` and not accepting any page parameter. 

```php
private static function _get_helper_redirect_url( $args = array() ) {
		global $current_screen;
		if ( isset( $_GET['redirect-to-wc-admin'] ) && $current_screen->id === 'woocommerce_page_wc-addons' ) {		
			return add_query_arg(
				array(
					'page' => 'wc-admin',
					'tab'  => 'my-subscriptions',
					'path' => urlencode( '/extensions' ),
				),
				admin_url( 'admin.php' )
			);
		}
		
		return add_query_arg(
			array_merge(
				array(
					'page'    => 'wc-addons',
					'section' => 'helper',
				),
				$args
			),
			admin_url( 'admin.php' )
		);
	}
```

So you can call it like:
```php
$redirect_uri = self::get_subscriptions_page_url(
			array(
				'filter'           => self::get_current_filter(),
				'wc-helper-status' => 'helper-refreshed',
			)
		);
```

Reply:
> I find it a little confusing that you instruct this function to redirect you to the wc-addons page when calling it

This is because WCCOM must redirect to wc-addons to finish the connection process.
If we would remove this part and just let it run in the new subscriptions page, we would need to copy all the connection code.

I found this to be an ok short term solution while we still have wc-addons and before we remove that page, we can move the connection code.

Would this be ok for now or would you like me to add full support for the connection flow to the new page?

Reply:
> Would this be ok for now or would you like me to add full support for the connection flow to the new page?

Absolutely. I forgot it's a short term solution while we support two pages at once. I'm happy to keep it as it is. 

>  If we would remove this part and just let it run in the new subscriptions page, we would need to copy all the connection code.
In case we refer back to this suggestion when we need to make a change, my suggestion actually keeps the current behavior. It's just written in a different way. It'll still redirect you to the right page (old or new). 

Anyway, thanks for the explanation @bgrgicak. I'll resolve this now.